src/OrchardCore.Modules/OrchardCore.OpenId/ViewModels/EditOpenIdApplicationViewModel.cs

```diff
@@ -30,13 +32,29 @@ public class EditOpenIdApplicationViewModel : IValidatableObject
         public bool AllowImplicitFlow { get; set; }
         public bool AllowLogoutEndpoint { get; set; }
 
-        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext) => validationContext.ValidateUrls(nameof(RedirectUris), RedirectUris)
-                .Union(validationContext.ValidateUrls(nameof(PostLogoutRedirectUris), PostLogoutRedirectUris));
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext) => ValidateUrls(validationContext, nameof(RedirectUris), RedirectUris)
+                .Union(ValidateUrls(validationContext, nameof(PostLogoutRedirectUris), PostLogoutRedirectUris));
 
         public class RoleEntry
         {
             public string Name { get; set; }
             public bool Selected { get; set; }
         }
+
+        private IEnumerable<ValidationResult> ValidateUrls(ValidationContext context, string memberName, string member)
```

Comment:
This will become hard to manage, once you removed the extension method

Reply:
I removed it as suggested by @sebastienros 

So now it is as in some other classes as `SmtpSettings`, resolved with the right type and by using a local `S` variable, so i removed the related field.

The only question i had is that maybe it would have been better to use a field for poextractor, if so we would have to update these other classes instead.